src/components/Notification/Notification.js

```diff
@@ -361,6 +390,7 @@ export class ToastNotification extends Component {
 
     return (
       <div {...other} role={role} kind={kind} className={classes}>
+        {NotificationIcon({ notificationType, kind, iconDescription })}
```

Comment:
Any reason for the eager evaluation? Could this be a component instead?

Reply:
no not without some test refactoring, which I'm willing to look into at a later time if that's alright

Reply:
@emyarod `<NotificationIcon>` seems to work as a component already in your PR and thus I guess @joshblack is suggesting `<NotificationIcon notificationType={notificationType} kind={kind} iconDescription={iconDescription} />`?

Reply:
yeah @asudoh  I was initially rendering the icon that way but our test suite was throwing errors which I didn't investigate closely yet

Reply:
yeah so as we discovered the node tree that Enzyme tests against changes when we render `<NotificationIcon {...props} />` instead of `NotificationIcon(props)`. I just went ahead and refactored the test suite to account for those changes